### DISCUSSION TITLE
Solana transfer throws "Invalid address string" error

### DISCUSSION THREAD

**Discussion by aberikashvili** [trustwallet/wallet-core#3733] [Q&A]
**Describe the bug**
I'm testing Solana transfer signer in `Swift`. I copy pasted example from https://github.com/trustwallet/wallet-core/blob/13cdf838f334a1101774c849ab82983b2e319701/swift/Tests/Blockchains/SolanaTests.swift#L31 to ensure it's working example, but I have obstacle.

My code looks like this:
```swift
@objc
  func solanaTest(_ name: String, phrase: String, callback: RCTResponseSenderBlock) {
    let privateKeyData = Data(Base58.decodeNoCheck( string: "A7psj2GW7ZMdY4E5hJq14KMeYg7HFjULSsWSrTXZLvYr")!)
    
    let transferMessage = SolanaTransfer.with {
        $0.recipient = "0x0987709642067c1449d288aeB2Cc9cA056F0D3f3"
        $0.value = 1
    }
    
    let input = SolanaSigningInput.with {
        $0.transferTransaction = transferMessage
        $0.recentBlockhash = "11111111111111111111111111111111"
        $0.privateKey = privateKeyData
    }

    let output: SolanaSigningOutput = AnySigner.sign(input: input, coin: .solana)
    
    callback([NSNull(), "{\"rawTransaction\":\"0x" + output.encoded + "\"}"])
  }
```
Calling `AnySigner.sign(input: input, coin: .solana)` method, throws the following error:
```
libc++abi: terminating due to uncaught exception of type std::invalid_argument: Invalid address string
```
And app crashes.

Any ideas? Any working docs?

**Comment by satoshiotomakan** [comment]
Hi @aberikashvili, the problem is that `Proto.SolanaTransfer. recipient` is a Ethereum-like address, but expected a Solana address.
